webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,185 @@
+import { STATUS } from 'foremanReact/constants';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { Bullseye, Button, Grid, GridItem } from '@patternfly/react-core';
+import { ChartDonut } from '@patternfly/react-charts';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { translate as __ } from 'foremanReact/common/I18n';
+import {
+  selectPublishContentViewsError, selectPublishContentViews,
+  selectPublishContentViewStatus,
+} from './ContentViewPublishSelectors';
+import { selectPublishTaskPoll, selectPublishTaskPollStatus } from '../Details/ContentViewDetailSelectors';
+import getContentViews, { publishContentView } from '../ContentViewsActions';
+import Loading from '../../../components/Loading';
+import EmptyStateMessage from '../../../components/Table/EmptyStateMessage';
+import { cvVersionPublishKey } from '../ContentViewsConstants';
+import { startPollingTask, stopPollingTask, toastTaskFinished } from '../../Tasks/TaskActions';
+import getContentViewDetails from '../Details/ContentViewDetailActions';
+
+
+const CVPublishFinish = ({
+  cvId,
+  userCheckedItems, setUserCheckedItems,
+  forcePromote, description, setDescription,
+  setIsOpen, versionCount, currentStep, setCurrentStep,
+}) => {
+  const dispatch = useDispatch();
+  const [publishDispatched, setPublishDispatched] = useState(false);
+  const [saving, setSaving] = useState(true);
+  const [polling, setPolling] = useState(false);
+  const [taskErrored, setTaskErrored] = useState(false);
+  const response = useSelector(state => selectPublishContentViews(state, cvId, versionCount));
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  const pollPublishTask = (cvPublishVersionKey, task) => {
+    if (!polling) dispatch(startPollingTask(cvPublishVersionKey, task));
+  };
+
+  const progressCompleted = () => (pollResponse.progress ? pollResponse.progress * 100 : 0);
```

Comment:
May as well move the `.toFixed` here

Reply:
I did refactor this a little:
progressPending method is nothing but 100 - progressCompleted() so using that..

toFixed() returns a string so can't used it in the chart as data directly except the title and label.